Workiva/ixbrl-viewer
# XT-2873: Units filter

#### Reason for change
Filter facts by complex units.

#### Description of change
This involved refactoring the current representation of "units" into "measures", and "units" now refers to the complex unit.

#### Steps to Test
Unit tests added.
Open doc and confirm expected results for:
- No unit selected
- Single unit selected
- Multiple units selected
- All units selected should match "numeric" filter

**review**:
@Workiva/xt
@paulwarren-wk
![image](https://github.com/Workiva/ixbrl-viewer/assets/105066394/1f9b5097-8e24-415b-99fd-8cfe63581390)
